llvm/lib/Target/DirectX/DXILRootSignature.cpp

```diff
@@ -241,6 +332,73 @@ static bool verifyRegisterSpace(uint32_t RegisterSpace) {
 
 static bool verifyDescriptorFlag(uint32_t Flags) { return (Flags & ~0xE) == 0; }
 
+static bool verifyRangeType(uint32_t Type) {
+  switch (Type) {
+  case llvm::to_underlying(dxbc::DescriptorRangeType::CBV):
+  case llvm::to_underlying(dxbc::DescriptorRangeType::SRV):
+  case llvm::to_underlying(dxbc::DescriptorRangeType::UAV):
+  case llvm::to_underlying(dxbc::DescriptorRangeType::Sampler):
+    return true;
+  };
+
+  return false;
+}
+
+static bool verifyDescriptorRangeFlag(uint32_t Version, uint32_t Type,
+                                      uint32_t FlagsVal) {
+  using FlagT = dxbc::DescriptorRangeFlag;
+  FlagT Flags = FlagT(FlagsVal);
+
+  const bool IsSampler =
+      (Type == llvm::to_underlying(dxbc::DescriptorRangeType::Sampler));
+
+  if (Version == 1) {
+    if (IsSampler)
+      return Flags == FlagT::NONE;
+    return Flags == FlagT::DESCRIPTORS_VOLATILE;
+  }
```

Comment:
These values are still incorrect. As I mentioned in https://github.com/llvm/llvm-project/pull/142492#discussion_r2140561978 samplers can only have DESCRIPTORS_VOLATILE, and otherwise this needs to be DATA_VOLATILE.

Also, please add a test for this - we should be catching that this is obviously wrong.